### ISSUE TITLE
arma 3 update to contact dlc on proton issues

### ISSUE THREAD

**Issue by osaris775** [muttleyxd/arma3-unix-launcher#43]
hi

sorry, this won't be technical but hopefully, you can figure it out and how to avoid it.

the update to the contact DLC breaks i think due to the symlinks created for the exe's by the arma3-unix-launcher. When i tried to launch Arma view either directly in steam or via the arma3-unix-launcher it failed. I had to copy the original exe's from arma3 from my windows pc to my Linux pc to fix it in which the arma3-unix-launcher now also works. 

if thats the case is there a way to avoid it like backing up the exe files and restoring them after the launcher exists?

thanks for your great work!

**Comment by muttleyxd** [comment]
Sorry for my late reply, but I didn't find much time recently

I created a PR with fix for this, A3UL won't copy/move any executables now, instead it'll use '-nolauncher' parameter

https://github.com/muttleyxd/arma3-unix-launcher/pull/44

Can you test it?

```
git clone https://github.com/muttleyxd/arma3-unix-launcher.git
git checkout do-not-mv-exes
mkdir -p build && cd build
cmake ..
make
./arma3-unix-launcher
```

**Comment by osaris775** [comment]
hi!

thanks! appears to work for me but unfortunately not sure if really fixed the issue for arma upgrades.  i do see that it creates a symlink of arma3launcher.exe to arma3_x64.exe. not sure if that's what borked it in the update to the contact dlc, cant remember exactly from the last time i looked at it.

 I deleted all the exe files for arma and replaced it with my windows arma exe files and then followed the steps you provided in your reply. I then ran your launcher again and I was able to launch the game fine with your launcher 

i attached the terminal output of when i launched it. doubt its useful but here you go.

[a3-unix-output.txt](https://github.com/muttleyxd/arma3-unix-launcher/files/3571750/a3-unix-output.txt)

Another thing unrelated to this was that i tried to get it to set the display to full screen windowed mode, any way to do that? the "force window mode" option uses the the windowed mode option of arma which is terrible. I tried using my own parameters file based what i have observed in the config file when selecting the options. This however might be wrong.

cat /home/osaris/Downloads/startup2.txt
-displayMode=2

displayMode=2 appears to be the fullscreen windowed mode (it changes it from 0 to 2 when changing from fullscreen to fullscreen windowed mode via the arma 3 display options and reflects in the Arma3.cfg file)

FYI - Some time ago i wrote a guide on how to setup arma 3 on proton and included instructions for you launcher. https://steamcommunity.com/sharedfiles/filedetails/?id=1755447470

thanks once again for your effort!

**Comment by muttleyxd** [comment]
> thanks! appears to work for me but unfortunately not sure if really fixed the issue for arma upgrades. i do see that it creates a symlink of arma3launcher.exe to arma3_x64.exe. not sure if that's what borked it in the update to the contact dlc, cant remember exactly from the last time i looked at it.

I got rid of all symlink creations regarding .exe files, so nothing is touched now. Only symlinks are created for !workshop mods

-----

> I deleted all the exe files for arma and replaced it with my windows arma exe files and then followed the steps you provided in your reply. I then ran your launcher again and I was able to launch the game fine with your launcher

Seems good, I'll merge the code to master. Unfortunately I prefer to not write code reverting the changes launcher has made.

-----

> 
> i attached the terminal output of when i launched it. doubt its useful but here you go.
> 
> [a3-unix-output.txt](https://github.com/muttleyxd/arma3-unix-launcher/files/3571750/a3-unix-output.txt)
Everything looks fine to me
```
class ModLauncherList
{
};
```
You were launching without any mods enabled, am I right?

-----

```
Another thing unrelated to this was that i tried to get it to set the display to full screen windowed mode, any way to do that? the "force window mode" option uses the the windowed mode option of arma which is terrible. I tried using my own parameters file based what i have observed in the config file when selecting the options. This however might be wrong.
```
I guess I should add an option to change this in launcher. Unfortunately there isn't an option for this now.

What you could try doing:
- start Arma without launcher
- change to fullscreen windowed mode 
- close Arma
- run launcher, run Arma through launcher
- check if Arma is in fullscreen window mode

-----

> thanks once again for your effort!

You're welcome

**Comment by muttleyxd** [comment]
Closed as stale, feel free to reopen if issue persists.

**Comment by muttleyxd** [comment]
It persists, reopening

**Comment by muttleyxd** [comment]
Fixed by 43d7c08, feel free to reopen if issue persists
